lib/model.js

```diff
@@ -3397,9 +3368,9 @@ class Model {
       } else {
         // Loop and call set
         if (options.attributes) {
-          let keys = options.attributes;
+          let keys = options.attributes.slice();
           if (this.constructor._hasVirtualAttributes) {
-            keys = keys.concat(this.constructor._virtualAttributes);
+            Utils.addSetToArray(keys, this.constructor._virtualAttributes);
```

Comment:
I didn't merged because this still doesn't looks right to me, I think we should keep using array for `_virtualAttributes`

Reply:
The point of this PR was to primarily to get rid of `memoize + includes` and replace with sets since that's slows all reads. Also keep in mind the set is only converted for iteration.

I could change the code in this location to not create arrays at all.

Reply:
> I could change the code in this location to not create arrays at all.

Hmm, give it a shot

Reply:
Done, waiting for build, I'm in the process of moving atm so I might take a while to reply.